segmentation/utils/segmentation_utils.py

```diff
@@ -16,7 +16,110 @@
 import skimage.io as io
 
 
-from segmentation.utils import plot_utils, signal_extraction
+from segmentation.utils import data_utils, plot_utils, signal_extraction
+
+
+def compute_complete_expression_matrices(segmentation_labels, base_dir=None, tiff_dir=None,
+                                         img_sub_folder="TIFs", is_mibitiff=False, mibitiff_suffix=None,
+                                         points=None, batch_size=5):
+    """
+    This function takes the segmented data and computes the expression matrices batch-wise
+    while also validating inputs
+
+    Inputs:
+        base_dir (str): the "master" directory, contains all of the data including the input
+        tiff_dir (str): the name of the directory which contains the single_channel_inputs,
+            technically can be accessed from base_dir but easier to make its own separate arg
+        img_sub_folder (str): the name of the folder where the TIF images are located
+        points (list): a list of points we wish to analyze, if None will default to all points
+        is_mibitiff (bool): a flag to indicate whether or not the base images are MIBItiffs
+        mibitiff_suffix (str): if is_mibitiff is true, then needs to be specified to select
+            which points to load from mibitiff
+        segmentation_labels (xarray): an xarray with the segmented data
+        batch_size (int): how large we want each of the batches of points to be when computing,
+            adjust as necessary for speed and memory considerations
+
+    Returns:
+        combined_normalized_data (pandas): a DataFrame containing the data with the size_norm transformation
+        combined_transformed_data (pandas): a DataFrame containing the data with the arcsinh transformation
+    """
+
+    if not base_dir:
+        raise ValueError("base_dir not specified")
+
+    if not tiff_dir:
+        raise ValueError("tiff_dir not specified")
+
+    if not img_sub_folder:
+        raise ValueError("img_sub_folder not specified")
+
+    # if no points are specified, then load all the points
+    if not points:
+        # if MIBItiff, only look at the points with the desired suffix...
+        if is_mibitiff:
+            # ...but of course, check if that suffix has been defined first
+            if not mibitiff_suffix:
+                raise ValueError("No mibitiff_suffix specified: please do so if loading from MIBItiff")
+
+            points = [point for point in points if point.endswith(MIBItiff_suffix)]
```

Comment:
Points hasn't been defined yet, I think this will fail. We should add a test for this

Reply:
Since we're removing MIBItiff_suffix, I think we can simply handle both the `is_mibitiff == True` and the `is_mibitiff == False` cases under the same `if points is None` case, so this line of code can be removed if I'm not mistaken.

Reply:
Actually, nvm. The mibitiff and tree structures are different from each other so I had to handle them separately.